utils/zookeeper-adjust-block-numbers-to-parts/main.cpp

```diff
@@ -28,36 +70,74 @@ size_t getMaxBlockSizeForPartition(zkutil::ZooKeeper & zk,
             }
             catch (const DB::Exception & ex)
             {
-                std::cerr << "Exception on: " << ex.displayText() << " will skip part: " << part << std::endl;
+                std::cerr << ex.displayText() << ", Part " << part << "skipped." << std::endl;
             }
         }
     }
     return max_block_num;
 }
 
-std::unordered_map<std::string, size_t> getAllTablesBlockPaths(zkutil::ZooKeeper & zk, const std::string & root)
+
+size_t getCurrentBlockNumberForPartition(zkutil::ZooKeeper & zk, const std::string & part_path)
+{
+    Coordination::Stat stat;
+    zk.get(part_path, &stat);
+
+    /// References:
+    /// https://stackoverflow.com/a/10347910
+    /// https://bowenli86.github.io/2016/07/07/distributed%20system/zookeeper/How-does-ZooKeeper-s-persistent-sequential-id-work/
+    return (stat.cversion + stat.numChildren) / 2;
```

Comment:
What if the user uses an older version of zookeeper? Or zookeeper unexpectedly change this rule?

Reply:
Maybe better to get a sequential node and parse number from its name?

Reply:
> Maybe better to get a sequential node and parse number from its name?

done